tests/testthat/test-query-search.R

```diff
@@ -960,3 +960,67 @@ test_that("can search for queries using boolean", {
     orderly_search(quote(parameter:a == "F"), root = root),
     character(0))
 })
+
+test_that("query with invalid types errors", {
+  root <- create_temporary_root(use_file_store = TRUE)
+  x <- create_random_packet(root, "x", list(a = TRUE))
+  y <- create_random_packet(root, "y", list(b = "test"))
+  z <- create_random_packet(root, "z", list(c = 2))
+
+  err <- expect_error(
+    orderly_search(quote(parameter:a > TRUE), root = root),
+    "Cannot use test '>' with supplied data type")
+  expect_equal(err$body,
+               c(x = "Search term 'parameter:a' of type 'logical' is invalid",
+                 x = "Literal value 'TRUE' of type 'logical' is invalid",
+                 i = "This test can be used with types:",
+                 "*" = "numeric"))
+
+  err <- expect_error(
+    orderly_search(quote(TRUE > parameter:a), root = root),
+    "Cannot use test '>' with supplied data type"
+  )
+  expect_equal(err$body,
+               c(x = "Literal value 'TRUE' of type 'logical' is invalid",
+                 x = "Search term 'parameter:a' of type 'logical' is invalid",
+                 i = "This test can be used with types:",
+                 "*" = "numeric"))
+
+  err <- expect_error(
+    orderly_search(quote(parameter:a > parameter:b), root = root),
+    "Cannot use test '>' with supplied data type"
+  )
+  expect_equal(err$body,
+               c(x = "Search term 'parameter:a' of type 'logical' is invalid",
+                 x = "Search term 'parameter:b' of type 'character' is invalid",
+                 i = "This test can be used with types:",
+                 "*" = "numeric"))
+
+  err <- expect_error(
+    orderly_search(quote(parameter:a > parameter:z), root = root),
+    "Cannot use test '>' with supplied data type"
+  )
+  expect_equal(err$body,
+               c(x = "Search term 'parameter:a' of type 'logical' is invalid",
+                 i = "This test can be used with types:",
+                 "*" = "numeric"))
+
+  expect_error(
+    orderly_search(quote(parameter:b >= "str"), root = root),
+    "Cannot use test '>=' with supplied data type"
+  )
+})
+
+test_that("query with invalid types errors even when one would satisfy", {
+  root <- create_temporary_root(use_file_store = TRUE)
+  x <- create_random_packet(root, "x", list(a = 2))
+  y <- create_random_packet(root, "y", list(a = "test"))
+
+  err <- expect_error(
+    orderly_search(quote(parameter:a > 1), root = root),
+    "Cannot use test '>' with supplied data type")
+  expect_equal(err$body,
+               c(x = "Search term 'parameter:a' of type 'character' is invalid",
+                 i = "This test can be used with types:",
+                 "*" = "numeric"))
```

Comment:
See this part, if two different reports use the same parameter with different types this is going to error. This won't be super consistent say if they did `name == "x" && parameter:a > 1` I think this will be fine if it does that index filtering on name first.

At the least we probably need to put the offending packet ID in which is causing this error to be thrown?

But maybe we should be more permissive.. not sure

Reply:
Fixed this up now, we'll no longer error but instead not use type coersion in the matching. So for the example above report y won't match but x will. No error will be thrown.